plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/PredicatePushdownController.java

```diff
@@ -30,9 +36,34 @@
         }
         return new DomainPushdownResult(domain, Domain.all(domain.getType()));
     };
-    PredicatePushdownController CASE_INSENSITIVE_PUSHDOWN = (session, domain) -> new DomainPushdownResult(
-            domain.simplify(getDomainCompactionThreshold(session)),
-            domain);
+
+    PredicatePushdownController CASE_INSENSITIVE_PUSHDOWN = (session, domain) -> {
+        checkArgument(
+                domain.getType() instanceof VarcharType || domain.getType() instanceof CharType,
+                "CASE_SENSITIVE_PUSHDOWN can be used only for chars and varchars");
+
+        DomainPushdownResult disabledPredicatePushdown = new DomainPushdownResult(Domain.all(domain.getType()), domain);
+
+        if (!domain.getValues().isDiscreteSet() && domain.getValues().complement().isDiscreteSet()) {
```

Comment:
This is already covered by the `for` below. Right?

Reply:
The one below handles any case where we have range which looks like `..., x)` or `(x,...`. Which is also true for case which is covered by this `if`. Unless I am missing something.